compiler/rustc_mir_transform/src/check_const_item_mutation.rs

```diff
@@ -53,9 +53,10 @@ impl<'tcx> ConstMutationChecker<'_, 'tcx> {
         //
         //     #[const_mutation_allowed]
         //     pub const LOG: Log = Log { msg: "" };
-        match self.tcx.adt_destructor(def_id) {
-            Some(_) => None,
-            None => Some(def_id),
+        match self.tcx.type_of(def_id).skip_binder().ty_adt_def().map(|adt| adt.has_dtor(self.tcx))
```

Comment:
This logic seems kinda sketch lol. Do we want to check that the type of the const `needs_drop` rather than just checking if it has a dtor directly?

Reply:
I guess it's preexisting.

Reply:
The change preseves the previous behaviour. If I do not change it we now never lint because a const DefId can never appear in an AdtDef

Reply:
I'll check if we want direct impls explicitly and leave a fixme if that's so or a comment otherwise